integrated_channels/canvas/client.py

```diff
@@ -411,11 +477,11 @@ def _handle_canvas_assignment_retrieval(self, integration_id, course_id):
         if not assignment_id:
             assignment_creation_data = {
                 'assignment': {
-                    'name': '(Edx integration) Final Grade',
```

Comment:
I know we used this name for final grades (and it is particularly relevant with Moodle), so:
How are we going to handle final grades in this scenario then? Are we just no longer doing a course-level grade if they use assignments?

Reply:
My hope was to create the assessment level reporting flow completely separated from course completion level reporting, while also leaving course level completely in tact. As it turns out, most of these hand shakes with Canvas are similar/identical whether doing assessment level or completion level (we create an assignment and post a submission to the assignment). So I tried and create common handlers that would take care of shared behaviors. [we are now passing the name of the "assignment"](https://github.com/edx/edx-enterprise/blob/771db5b8391249e57313966e3f0432d4a6e24fe0/integrated_channels/canvas/client.py#L418) (name of the edx subsection or the name "final grade") to the helper function. [Here is where we're passing the "final grade" name into the assignment creation process](https://github.com/edx/edx-enterprise/blob/771db5b8391249e57313966e3f0432d4a6e24fe0/integrated_channels/canvas/client.py#L208) so both should still work!

Reply:
This also has the added benefit of being more easily testable without having to stub out all previous steps in the process! We can just call the helper method and test that it behaves properly.

Reply:
Nice. I missed that addition.